### ISSUE TITLE
Response body stream from SW has wrong realm

### ISSUE THREAD

**Issue by jyasskin** [whatwg/fetch#731]
If I'm tracing this correctly, a [`fetch()`](https://fetch.spec.whatwg.org/#fetch-method) from a Window that's handled by a Service Worker returns a Response object in the Window's realm associated with the [response struct](https://fetch.spec.whatwg.org/#concept-response) created by the Service Worker and returned via https://fetch.spec.whatwg.org/#http-fetch. That response struct contains a [body](https://fetch.spec.whatwg.org/#concept-body) with a stream containing a `ReadableStream` object created in https://w3c.github.io/ServiceWorker/#fetch-event-respondwith that's associated with the Service Worker's realm.

https://w3c.github.io/ServiceWorker/#fetch-event-respondwith copies the stream, and w3c/ServiceWorker#934 says it's trying to do this exactly to get the realms right, but it uses *targetRealm* defined as "relevant Realm of the context object" of the FetchEvent, which is just the Service Worker's realm again, not the `fetch()` call's realm. I believe this copy needs to be done inside Fetch in order to have the right realm available, not inside the Service Worker spec. (@yutakahirano)

This might be a sign that it's a mistake to have the body struct contain a Javascript object, but streams don't provide a non-Javascript abstraction for the body to contain instead. (@domenic)

**Comment by annevk** [comment]
Yeah, this (and the other issue) strongly argue for an abstract stream being available.

**Comment by annevk** [comment]
@domenic @jakearchibald @ricea thoughts on how to handle this?

**Comment by ricea** [comment]
In this specific case I think it would make sense to say the stream is cloned and then the clone is [transferred](https://streams.spec.whatwg.org/#rs-transfer) to the appropriate realm.

This doesn't solve the general problem of being able to talk about streams that aren't tied to a specific realm.

**Comment by annevk** [comment]
@yutakahirano this relates to that other issue you are working on with regards to how Fetch and Service Workers should integrate when it comes to streams.
